modules/products/confluence/locals.tf

```diff
@@ -87,4 +87,55 @@ locals {
       ]
     }
   }) : yamlencode({})
+
+  # After restoring the snapshot of the Confluence database, a re-index is required. An alternative is recover the index.
+  # Because there is no node existed when the environment is created, so index recovery cannot be borrow the files from other nodes
+  # and is forced to use the recovery journal files from shared home.
+  # By default recovery index files are only valid for 2 days which is not enough in our use case.
+  # One way for extend the validity duration for index recovery file is override the `com.atlassian.confluence.journal.timeToLiveInMillis`
+  # system property. Also we may need to extend the index recovery timeout as for enterprise data we may hit the timeout and it will
+  # results a failure in index recovery and Confluence will start a full re-index which is not efficient.
+  # For more info see the following links:
+  # https://confluence.atlassian.com/conf78/recognized-system-properties-1021242818.html
+  # https://extranet.atlassian.com/pages/viewpage.action?pageId=4215563160
+  # https://extranet.atlassian.com/pages/viewpage.action?spaceKey=CONFARCH&title=Index+recovery
+
+  # Current date - the date the index recovery files are going to be used
+  date_year  = tonumber(formatdate("YYYY", timestamp())) # current year
+  date_month = tonumber(formatdate("MM", timestamp())) # current year
+  date_days  = tonumber(formatdate("DD", timestamp())) # current year
+  day_millis = 24 * 360 * 10000 # number of milliseconds in a day
+  # Get the snapshot creation date
+  creation_date  = split("-", var.shared_home_snapshot_creation_date)
+  snapshot_year  = tonumber(local.creation_date[0])
+  snapshot_month = tonumber(local.creation_date[1])
+  snapshot_day   = tonumber(local.creation_date[2])
+  # Calculate the days passed from snapshot creation + 7 days extra
+  offset       = ((local.snapshot_month - 1 ) * 30) + local.snapshot_day
+  days_from_snapshot = (local.date_year - local.snapshot_year) * 365 + local.date_month * 30 + local.date_days - local.offset + 7
+
+  # Override the lifetime of recovery index in shared home and make sure there is enough timeout for copy long files
+  extend_snapshot_validity = var.db_snapshot_id != null ? yamlencode({
+    confluence = {
+      additionalJvmArgs = [
+        "-Dcom.atlassian.confluence.journal.timeToLiveInMillis=${local.days_from_snapshot * local.day_millis}", # milliseconds
+        "-Dconfluence.cluster.index.recovery.generation.timeout=480", # seconds
+        "-Dconfluence.cluster.snapshot.file.wait.time=480", # seconds
+      ]
+    }
+  }) : yamlencode({})
+
+  number_of_threads = min(4, floor(tonumber(local.confluence_software_resources.cpu)))
```

Comment:
This assumes we always get an integer (not `2500m` for example). The `confluence_cpu` is defined as a `string` type so it is possible to pass it in a different format. We can potentially make it more explicit, convert it here or at least document it.